src/ex12_js-dom/task.js

```diff
@@ -0,0 +1,69 @@
+var tags={};
+var classes={};
+tags.text=0;
+function scanDOM(elem){
+ for (var i = 0; i < elem.childNodes.length; i++) 
+ {
+  if(elem.childNodes[i].nodeType === 1)
+	{
+	  if(!!elem.childNodes[i].classList)
+	  {
+		for (var j=0; j<elem.childNodes[i].classList.length; j++)
+		{
+			if (elem.childNodes[i].classList[j] in classes)
+			{
+			 classes[elem.childNodes[i].classList[j]]++;	
+			}
+			else
+			{
+			 classes[elem.childNodes[i].classList[j]]=1;	
+			}
+		}
+	  }
+	  if (elem.childNodes[i].localName in tags)
```

Comment:
Два похожих блока. Надо отрефакторить в один.

Reply:
Не могу понять, как это можно сделать. По сути, я же в цикле перебираю всех детей текущего узла. Для каждого дочернего элемента просматриваю и считаю все его классы:
```suggestion
 for (var j=0; j<elem.childNodes[i].classList.length; j++)
 {
  if (elem.childNodes[i].classList[j] in classes)
  {
   classes[elem.childNodes[i].classList[j]]++;	
  }
  else
  {
   classes[elem.childNodes[i].classList[j]]=1;	
  }
 }
```
потом учитываю сам элемент:
```suggestion
 if (elem.childNodes[i].localName in tags)
 {
  tags[elem.childNodes[i].localName]++;	
 }
 else
 {
  tags[elem.childNodes[i].localName]=1;
 }
```
 2 этих действия объединить в 1 без увеличения размера  кода у меня не получается.
Единственное, убрал проверку на непустой classList:
```suggestion
 if(!!elem.childNodes[i].classList)
```
потому что при пустом classList цикл просто не запустится и смысла от данной проверки нет

Reply:
```
function increaseCounter(store, key) {
    if (key in store)
    {
        store[key]++;	
    }
    else
    {
        store[key]=1;	
    }
}

increaseCounter(classes, elem.childNodes[i].classList[j]);
```

Reply:
Это оказалось так просто, что я теперь чувствую себя тупеньким >_< 
Единственное, пришлось сделать так: 
`var storage=store;`
А то линтер ругается на изменение параметров функции